relay-server/src/utils/dynamic_sampling.rs

```diff
@@ -72,6 +72,12 @@ fn get_trace_sampling_rule(
     let sample_rate = match sampling_config.mode {
         SamplingMode::Received => rule.sample_rate,
         SamplingMode::Total => sampling_context.adjusted_sample_rate(rule.sample_rate),
+        SamplingMode::Unsupported => {
+            if processing_enabled {
+                relay_log::error!("Found unsupported sampling mode even as processing Relay, keep");
+            }
+            return Err(SamplingResult::Keep);
```

Comment:
I'm not sure if I like this, but I don't have better suggestions.

Reply:
We could alternatively log an error for during project state deserialization (since that's where the error really happens) and fail silently here instead. 

This behavior is fine since it's the same as if sampling were not implemented. That matches behavior in other cases, where Relay ignores all configuration it doesn't know about.

Reply:
> We could alternatively log an error for during project state deserialization (since that's where the error really happens)

I'd rather log the error during deserialization.

Reply:
I tried to be consistent with `check_unsupported_rules` here, but maybe it's a bad pattern anyway.

https://github.com/getsentry/relay/blob/ca275954015fb23d5f9c96bc55f3db29435f660a/relay-server/src/utils/dynamic_sampling.rs#L33-L47

Reply:
As discussed offline, going to merge as-is.